### ISSUE TITLE
Pre-submission enquiry

### ISSUE THREAD

**Issue by zoj613** [openjournals/joss#941]
Hello,

A while ago I wrote a python package to facilitate efficient sampling from a Polya-Gamma distribution in a Numpy/SciPy compatible way. It turned out to be more efficient than all existing implementations I am currently aware of. I also ended up adding functionality to compute the CDF, density and their logarithms. A few days ago someone [asked about citing the software](https://github.com/zoj613/polyagamma/issues/82) in their research, which prompted me to think about whether or not I should write up a documentation paper for such purposes. A friend pointed me to this journal, but before anything else I wanted to inquire if such a package is worthwhile submitting here?

Regards

**Comment by jedbrown** [comment]
I take it the C library was implemented independently (no copyright/license issues with respect to pypolyagamma or other libraries). It looks in-scope to me. One consideration is whether it is better as a stand-alone package or contributed as a module to a library like statsmodels. (It's possible to submit significant modules of existing packages to JOSS.)

**Comment by zoj613** [comment]
> I take it the C library was implemented independently (no copyright/license issues with respect to pypolyagamma or other libraries). It looks in-scope to me. One consideration is whether it is better as a stand-alone package or contributed as a module to a library like statsmodels. (It's possible to submit significant modules of existing packages to JOSS.)

No, it does not copy the GPL code wrapped by pypolyagamma. It is a clean room implementation based on the algorithm details from the following papers:
1) https://arxiv.org/abs/1205.0310
2) https://arxiv.org/abs/1405.0506

I first proposed contributing an early version of the implementation to Numpy (see https://github.com/numpy/numpy/issues/18076), then Scipy and statsmodels (see: https://github.com/scipy/scipy/issues/11009). All the aforementioned chose not to decline the contribution because they deemed it too involved (complicated) as a single contribution and was better as a standalone library. Hence why I ended up writing it after that feedback.
